### ISSUE TITLE
Collect Responses not in Type Definition

### ISSUE THREAD

**Issue by pducks32** [howdyai/botkit#1356]
Hi was using (and loving) botkit when I wanted to move to typescript. Everything was doing great until I couldn’t find the function `collectResponse`.
As defined here
https://github.com/howdyai/botkit/blob/0596455b04c8cfedc801057b41d4b8da3bb0cf40/lib/CoreBot.js#L181

It’s not listed in the type signatures. 
https://github.com/howdyai/botkit/blob/0596455b04c8cfedc801057b41d4b8da3bb0cf40/lib/Botkit.d.ts#L128

I recognize this function shouldn’t be used publically but I have a really important use/edge case where I need it. Would it be acceptable to add to the type definitions if I made a PR?

**Comment by Naktibalda** [comment]
@pducks32 Please make a pull request.

**Comment by peterswimm** [comment]
"I recognize this function shouldn’t be used publically but I have a really important use/edge case where I need it. Would it be acceptable to add to the type definitions if I made a PR?"

If its to fix a small edge case, any other reason to make this part of the default experience for Botkit versus just maintaining your own fork? Just trying understand more the reasoning behind the fix.

**Comment by pducks32** [comment]
Well the function is public in the javascript so I feel like it's there for use. Since this is a library I personally take the belief that the consumer should be allowed to use anything that's public and in that sense the types should reflect what a consumer of the JS library would expect. 

So though I have to use it for my use case, the reason I used it is because when my bot was written in JS it was a public method so I wrote this issue because I just assumed it was an oversight in writing the `d.ts` file. I might be wrong and that function is supposed to be private, but then I'd argue that there should be a really good reason why it's private.
